qtpy/__init__.py

```diff
@@ -78,13 +78,13 @@ class PythonQtValueError(ValueError):
 
 class QtModuleNotFoundError(ModuleNotFoundError, PythonQtError):
     """Raised when a Python Qt binding submodule is not installed/supported."""
-    _msg = 'The {name} module was not found{binding}.'
-    _msg_binding = ' for {binding}'
+    _msg = 'The {name} module was not found.'
+    _msg_binding = '{binding}'
     _msg_extra = ''
 
-    def __init__(self, *, name, binding=None, msg=None, **msg_kwargs):
-        self.binding = binding
-        binding = self._msg_binding.format(binding=binding) if binding else ''
+    def __init__(self, *, name, msg=None, **msg_kwargs):
+        global API_NAME
+        binding = self._msg_binding.format(binding=API_NAME)
```

Comment:
Using global _can_ be problematic, but in this case it's only reading a file scope variable and not writing to it. If this is still undesirable, I do have an alternate idea to write the same feature.

Reply:
Could you share the alternative idea please?

Reply:
(I spent 20 minutes trying to insert the proposal into a suggestion box but I had to give up. GitHub has too many buttons.)


> Could you share the alternative idea please?

It's effectively just the inverse of the first version.

This assigns the selected `API_NAME` as a class attribute of `QtModuleNotFoundError`, so the value will already be set whenever one of these Error objects is created. The definition of `QtModuleNotFoundError` becomes this:

```py
class QtModuleNotFoundError(ModuleNotFoundError, PythonQtError):
    """Raised when a Python Qt binding submodule is not installed/supported."""
    _msg = 'The {name} module was not found.'
    _msg_binding = '' # this attribute is assigned when API_NAME is identified
    _msg_extra = ''

    def __init__(self, *, name, msg=None, **msg_kwargs):
        msg = msg or f'{self._msg} {self._msg_extra}'.strip()
        msg = msg.format(name=name, binding=_msg_binding, **msg_kwargs)
        super().__init__(msg, name=name)

# <snip>
# line ~255:

# Set display name of the Qt API
API_NAME = API_NAMES[API]

# set the selected binding for the error classes
QtModuleNotFoundError._msg_binding = API_NAME
```

Personally, I think this is worse than using `global` because this 'feature' is spread out across 200 lines. In the original, you only need to read the body of `QtModuleNotFoundError` to see how it works.

Reply:
Thanks for taking the time to explain it @DaelonSuzuka ! From my side then I would say `global` is better. What do you think @CAM-Gerlach @ccordoba12 ? Maybe someone has another alternative idea to achieve the same behavior?

Reply:
> From my side then I would say `global` is better. What do you think @CAM-Gerlach @ccordoba12 ?

Yep, `global` seems simpler.